UnitsNet/CustomCode/Quantities/Length.extra.cs

```diff
@@ -236,5 +236,70 @@ public string ToString(IFormatProvider? cultureInfo)
             // So inches are rounded when converting from base units to feet/inches.
             return string.Format(cultureInfo, "{0:n0} {1} {2:n0} {3}", Feet, footUnit, Math.Round(Inches), inchUnit);
         }
+
+        /// <summary>
+        ///     Outputs feet and inches on the format: {feetValue}' - {inchesValueIntegral}[ {inchesValueFractional}]"
+        ///     The inches are rounded to the nearest fraction of the fractionDenominator argument and reduced over the greatest common divisor.
+        /// </summary>
+        /// <param name="fractionDenominator">The maximum precision to express the rounded inch fraction part</param>
+        /// <example>Length.FromFeetInches(3,2).ToArchitecturalString(32).ToString() outputs: "3 ft 2 in"</example>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when fractionDenominator is &lt; 1 (use 1 to round to nearest integer, with no fraction)</exception>
+        public string ToArchitecturalString(int fractionDenominator)
+        {
+            if (fractionDenominator < 1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(fractionDenominator), "Denominator for fractional inch cannot be zero");
```

Comment:
To be more precise, maybe change it to `"Denominator for fractional inch must be greater than zero."`

Reply:
You could also put this text in the `ArgumentOutOfRangeException` xmldoc.